src/Ledger/Gov.lagda

```diff
@@ -99,42 +99,89 @@ open PState
 \begin{figure*}
 \begin{AgdaMultiCode}
 \begin{code}
-govActionPriority : GovAction → ℕ
-govActionPriority NoConfidence             = 0
-govActionPriority (UpdateCommittee _ _ _)  = 1
-govActionPriority (NewConstitution _ _)    = 2
-govActionPriority (TriggerHF _)            = 3
-govActionPriority (ChangePParams _)        = 4
-govActionPriority (TreasuryWdrl _)         = 5
-govActionPriority Info                     = 6
-
-Overlap : GovAction → GovAction → Type
-Overlap a a' = let n = govActionPriority a
-                   m = govActionPriority a'
-               in (n ≡ m) ⊎ (n ≡ 0 × m ≡ 1) ⊎ (n ≡ 1 × m ≡ 0)
+govActionPriority : GovActionType → ℕ
+govActionPriority NoConfidence     = 0
+govActionPriority UpdateCommittee  = 1
+govActionPriority NewConstitution  = 2
+govActionPriority TriggerHF        = 3
+govActionPriority ChangePParams    = 4
+govActionPriority TreasuryWdrl     = 5
+govActionPriority Info             = 6
+
+Overlap : GovActionType → GovActionType → Type
+Overlap NoConfidence    UpdateCommittee  = ⊤
+Overlap UpdateCommittee NoConfidence     = ⊤
+Overlap a               a'               = a ≡ a'
 \end{code}
 \begin{code}[hide]
-Overlap? : (a a' : GovAction) → Dec (Overlap a a')
-Overlap? a a' = let n = govActionPriority a
-                    m = govActionPriority a'
-                in n ≟ m ⊎-dec (n ≟ 0 ×-dec m ≟ 1) ⊎-dec (n ≟ 1 ×-dec m ≟ 0)
+-- TODO: cleanup this
```

Comment:
This has come up before in a similar scenario on another project, and I think the answer is to make the `match` tactic (from `Tactic.Match` in stdlib-meta) work properly. Then you could just write something like `Overlap? = by $ match $ match $ anyOf (quote (Dec-⊤ .dec) ∷ quote (yes refl) ∷ quote (no (λ ())) ∷ [])`.

Reply:
Opened an [issue](https://github.com/agda/agda-stdlib-meta/issues/27) in agda-stdlib-meta regarding this.